pkg/types/interfacetests/utils.go

```diff
@@ -36,12 +41,88 @@ func runTests[T TestingT[T]](t T, tester BasicTester[T], tests []testcase[T]) {
 	}
 }
 
+func batchChainWrite[T TestingT[T]](t T, tester ChainComponentsInterfaceTester[T], batchCallEntry BatchCallEntry, mockRun bool) {
+	if mockRun {
+		cw := tester.GetChainWriter(t)
+		err := cw.SubmitTransaction(tests.Context(t), AnyContractName, "batchChainWrite", batchCallEntry, "", "", nil, big.NewInt(0))
+		require.NoError(t, err)
+		return
+	}
```

Comment:
Why is this needed?  The mock is a good test for the relay so it's nice when realish data goes over it.  It makes it more likely to catch serialization errors earlier on.

Reply:
This is needed because in the mocked tests, we have to use the helper function [SetBatchLatestValues](https://github.com/smartcontractkit/chainlink-common/blob/b10e2708ab783ff269fac0d98a0dfc1624649c8c/pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader_test.go#L407-L414) which accepts a `batchCallEntry`. When we aren't doing a mockRun, we have to send individual transactions sequentially since `ChainWriter` has no batch send functionality atm.

Reply:
Is this resolved now? What Silas said seems reasonable to me.